### ISSUE TITLE
Falsely detected as HackTool:Win64/ExplorerPatcher!MTB

### ISSUE THREAD

**Issue by Comeonnoob** [valinet/ExplorerPatcher#3228]
For some reason, latest release of EP keeps getting false flagged by Windows defender as HackTool:Win64/ExplorerPatcher!MTB. I have to exclude this app's folder from Program Files manually to uninstall it properly. What is going on?

![изображение](https://github.com/valinet/ExplorerPatcher/assets/140257029/328851e2-93f6-4223-baf8-834a7b882909)

**Comment by Amrsatrio** [comment]
Microsoft does not like EP anymore, seems like. It's understandable since they've been removing legacy stuff which EP resurrects. Adding to exclusions or compiling your own EP seems to be the only way now.

**Comment by pyrates999** [comment]
You can also set windows defender to exclude C:\Program Files\ExplorerPatcher so future updates won't be blocked by windows defender.
You can also set windows defender to exclude the directory that you manually download EP to so you can install it without windows defender blocking it.

**Comment by Amrsatrio** [comment]
Also %APPDATA%\ExplorerPatcher

**Comment by Vlad55432** [comment]
microsoft is so mad ngl

**Comment by Apis035** [comment]
Thats not a false detection. They literally named it as "HackTool:Win64/ExplorerPatcher!MTB" because they don't like EP.

**Comment by kpietraszko** [comment]
It would be nice if EP deleted ep_setup.exe as soon as possible after an update, that would probably decrease the detection rate.

**Comment by pyrates999** [comment]
It's used for uninstalling.  It can't be deleted before then.

**Comment by Comeonnoob** [comment]
> You can also set windows defender to exclude C:\Program Files\ExplorerPatcher so future updates won't be blocked by windows defender. You can also set windows defender to exclude the directory that you manually download EP to so you can install it without windows defender blocking it.
Thank you, had to use this

Also it looks like Windows 11 24H2 update might be a disaster because Microsoft is actively blocking StartAllBack as well. Not good, Microsoft needs lawsuit from someone with a decent anmount of money.

**Comment by Amrsatrio** [comment]
Just want to post an update to this issue, looks like this will continue to haunt us forever even with local builds.

I am planning to make a PowerShell-based online installer which will be a simple GUI to select between release and pre-release version as well as optionally including the folders into exclusions which will be checked by default. And when they flag the PowerShell script, we can easily break the hash by modifying the script more like what [MAS has been doing monthly](https://github.com/massgravel/Microsoft-Activation-Scripts/commit/d3f2cd88061ff66571628ea2342be21e6cd8ae64).

For updates, users will need to download updates through EP itself not by visiting the GitHub releases page anymore.

What do you guys think?

**Comment by Vlad55432** [comment]
> What do you guys think?

Powershell script is nice touch, I like that.

**Comment by pyrates999** [comment]
great idea

**Comment by Anixx** [comment]
It is not a false detection, it correctly detects it as hacking tool ExplorerPatcher.

**Comment by rodolfomachado** [comment]
Hi, guys.
Ive been using EP for a good time with no issues but today when I started my PC Kaspersky detected as a dangerous object and deleted my Explorer.exe. My PC is starting but with almost no interface. I can use task manager, command. For now, I was not able to restore Explorer.exe.
Does anyone know a way? Tks

**Comment by Kickskii** [comment]
its marked as Backdoor:Win32/Bladabindi!ml now earlier today it was calling it HackTool:Win32/Patcher!MTB

**Comment by Anixx** [comment]
Unlike most Trojan, Backdoor:Win32/Bladabindi!ml does not create a registry entry to run itself on Windows start-up. Instead, this threat will inject harmful code into valid processes including explorer.exe, iexplore.exe, firefox.exe, chrome.exe, opera.exe, and safari.exe. Trojan will load if user runs any of these programs.

Then, the Trojan tries to contact a command and control (C&C) server through HTTP request on the same port 80, the same way users can connect to the Internet. During analysis, it was discovered that most of C&C servers that provides remote command for this threat are originating from .TW domains.

Lastly, Backdoor:Win32/Bladabindi!ml attempts to gather cookie data from the infected computer. It is also interested in collecting Internet certificates and stores them under UserProfile folder.

**Comment by rodolfomachado** [comment]
Here is what Kaspersky shows

![image](https://github.com/user-attachments/assets/1661ef8c-a7e9-4660-af69-456b7b74dbec)

**Comment by arnileibovits** [comment]
If you're having problems downloading, you can add the downloadable `ep_setup.exe` to the exclusions list:
```
Add-MpPreference -ExclusionPath "%UserProfile%\Downloads\ep_setup.exe"
```

**Comment by piterq** [comment]
I see that previous setup file size was around 2 megabytes (checked on versions: 22621.2506.60.1 and 22621.3527.65.5). Setup file size for version 22621.3880.66.5 is around 10 megabytes. Windows Defender is blocking only this new version - older versions are not affected. Strange.

**Comment by ItsmcFox** [comment]
You guys are actually using microsoft defender? I use windows security and got the same threat detections.

**Comment by perdrix52** [comment]
Defender says Backdoor:Win32/Bladabindi!ml  This isn't the issue previously reported.  It only started with the very latest update

**Comment by Amrsatrio** [comment]
@piterq Yes, only on 66.4 I started including the Windows 10 start menu binaries from the last 22621 build that has it. As well as the reimplemented Windows 10 taskbar because without them this app would be useless on 24H2. Blame Microsoft for nuking them; we did what was needed to maintain functionality.

**Comment by pyrates999** [comment]
> If you're having problems downloading, you can add the downloadable `ep_setup.exe` to the exclusions list:
> 
> ```
> Add-MpPreference -ExclusionPath "%UserProfile%\Downloads\ep_setup.exe"
> ```

You should be using a subdirectory inside your downloads directory just for ep_setup.exe.  You've just excluded everything you download from the internet by doing that.  Such as:

`Add-MpPreference -ExclusionPath "$env:UserProfile\Downloads\ExplorerPatcher\ep_setup.exe"`

**Comment by leighss** [comment]
> Hi, guys. Ive been using EP for a good time with no issues but today when I started my PC Kaspersky detected as a dangerous object and deleted my Explorer.exe. My PC is starting but with almost no interface. I can use task manager, command. For now, I was not able to restore Explorer.exe. Does anyone know a way? Tks
> 
> Edit: Solved with "sfc /scannow

I tried disabling BitDefender antivirus for update after it kept saying it was disinfecting, and the update started but the BitDefender jumped in and I was left with no interface (I am now  in Linux)
I have same problem now, lock screen in Win 11 working but when I log in I get black screen with just mouse cursor (working).
Gonna try to open task manager with Ctrl+Alt+Delete and/or Win+X

What does "sfc /scannow" do?

**Comment by pyrates999** [comment]
"sfc /scannow" repairs your windows install.

**Comment by leighss** [comment]
```
C:\Windows\System32>sfc /scannow

Beginning system scan.  This process will take some time.

Beginning verification phase of system scan.
Verification 100% complete.

Windows Resource Protection did not find any integrity violations.
```

I tried to uninstall ExplorerPatcher via Revo Uninstaller, but it said the uninstallcommand was missing

I tried to download ep-setup.exe from here to try to use that, and of course BitDefender Blocked that

**Comment by leighss** [comment]
"sfc /scannow" found nothing
Revo uninstall failed to uninstall ExplorerPatcher (since Bitdefender had disinfected updat / installer file?)
I obviously couldnt download installer file
Made sure I disactivated all aspects of Bit Defender
Managed to download installer file
Tried to run it but error that parts of previous version still open
Tried all suggested options, then rebooted and  I am now on the latest release 22621.3880.66.5
I will check if my copy paste etc in File explorer work before making decision whether or not to uninstall
But it seems that MS now has its guns on ExplorerPatcher :(

I wonder what the real reason MS doesnt want the taskbar at the top?
Shooting down 3rd party apps is understandable (only in the sense of knowing why, not condoning AT ALL!), 
but why not just allow people to move it to the top? 
Must be something more sinister / financial than just pig-haded ignorance and incompetence IMO

**Comment by pyrates999** [comment]
MS just want to show you ads in windows.  That's why they want you to use windows 11 taskbar and start menu.

> I tried to download ep-setup.exe from here to try to use that, and of course BitDefender Blocked that

You need to exclude all the directories specified here in Bitdefender: https://github.com/valinet/ExplorerPatcher/releases

Then install EP.

**Comment by leighss** [comment]
Of course! Ads!
Just like any search (via Google only?) now only returns commercial (whether obviously so or disguised) results.

I still had freezing etc problems in File Explorer, so decided to uninstall Explorer Patcher sadly

I will now move to not using Windows for Work as I havent done for years at home
I'm an academic and unfortunately writing paper is done by nearly all co-author's in Word

But I have invested in SoftMaker Office NX, but havent had the time to invest in getting familiar with it.
That's my next task
As far as all other APPS, (and I think Softmaker too) the Linux alternatives are just better - No need at all for Windows, and Linux OS's (Even Arch Based ones , e.g. Endeavour OS) are just light years ahead and very simple to install and maintain these days. My 80 year old techno-turnip mum has been on EOS for years now, without problems

**Comment by Clydefrosch22** [comment]
> > If you're having problems downloading, you can add the downloadable `ep_setup.exe` to the exclusions list:
> > ```
> > Add-MpPreference -ExclusionPath "%UserProfile%\Downloads\ep_setup.exe"
> > ```
> 
> You should be using a subdirectory inside your downloads directory just for ep_setup.exe. You've just excluded everything you download from the internet by doing that. Such as:
> 
> `Add-MpPreference -ExclusionPath "$env:UserProfile\Downloads\ExplorerPatcher\ep_setup.exe"`

do you happen to know how to undo  exclusions? I'm not sure if i did exclude the full downloads earlier but just on case

**Comment by plato1123** [comment]
Wow being forced (momentarily) back to Windows 11 default start menu is really like a slap in the face.  Why would Microsoft think such a terrible start menu experience might be ok?  How is Microsoft still in denial about this?  Surreal.  

(Thanks to the Explorer Patcher team!!)

**Comment by pyrates999** [comment]
> > > If you're having problems downloading, you can add the downloadable `ep_setup.exe` to the exclusions list:
> > > ```
> > > Add-MpPreference -ExclusionPath "%UserProfile%\Downloads\ep_setup.exe"
> > > ```
> > 
> > 
> > You should be using a subdirectory inside your downloads directory just for ep_setup.exe. You've just excluded everything you download from the internet by doing that. Such as:
> > `Add-MpPreference -ExclusionPath "$env:UserProfile\Downloads\ExplorerPatcher\ep_setup.exe"`
> 
> do you happen to know how to undo exclusions? I'm not sure if i did exclude the full downloads earlier but just on case.
> 
> also so far, its still not letting me install anything.
> 
> i'm using windows security and i guess the powershell command doesnt work for me?
> 
> Add-MpPreference : Fehler beim Vorgang: 0x%1!x! In Zeile:1 Zeichen:1
> 
> * Add-MpPreference -ExclusionPath "C:\Program Files\ExplorerPatcher"
> * ```
>     + CategoryInfo          : NotSpecified: (MSFT_MpPreference:root\Microsoft\...FT_MpPreference) [Add-MpPreference],
>    CimException
>     + FullyQualifiedErrorId : HRESULT 0xc0000142,Add-MpPreference
>   ```

That error indicates you are running another AV and not windows defender.  You must exclude the directories specified in the AV you're running.

**Comment by joeysirmons** [comment]
Just wanted to chime in on this thread. I've been using ExplorerPatcher without issue for quite some time until today, when I tried to update to version 22621.3880.66.5. Windows Security flagged it as malicious and quarantined it. Fine, Whatever. That happens sometimes.

So I came to GitHub to download it directly. Quarantined again. So I downloaded again and prevented quarantine and then did what I always do -- uploaded the executable to VirusTotal.

So Microsoft doesn't like ExplorerPatcher. I get that and I'm completely willing to accept that at face value. Due to the nature of EP, I could even accept 3 or 4 AV engines flagging it as questionable. But I'm seeing 32 out of 68 virus engines flagging it as outright malicious.
[VirusTotal results](https://www.virustotal.com/gui/file/1c4e1847c722db18d58216c43aa40ad87c8a38aa6196e69d55c0687b8506bf94/detection)

I really, genuinely appreciate the work you've done on this application, but I have to weigh the risks to my network. When nearly half of the AV engines on VT flag something -- that's just more risk than I'm comfortable with.

For those who would say I've trusted it this far -- yes, I have. Because it hasn't been flagged before today. And now it is being flagged. So what changed?

I am not a programmer, but I have some small grasp on high-level programming concepts. I would love to be educated and learn why I should ignore this many warnings. I would appreciate any constructive replies.

**Comment by Amrsatrio** [comment]
@joeysirmons Okay let me explain my speculations and what I've done.

These HackTool detections all started in February after the news that Satya emphasized "security, security, security" after a series of horrible data breaches.

This also began with the crippling of the Windows 10 start menu in early 24H2 builds aka 25xxx builds. After some months the breakages came to 22H2 and 23H2 which raised my attention. This broke EP's original method for enabling the 10 start menu by Valinet which required me to rearchitect the method, which shipped with 65.x or something. For context 66.5 slipped under Microsoft's radar hence why I was able to put that as the release version for 3 months.

And just 2 months ago they crippled the start menu even more in 24H2 build 26224. They removed StartUI.dll and its relevant resources in the pri (resource) files so that they need to be restored. This led me into the need bundling the related files into ep_setup.exe in order to keep the old start menu working on affected versions.

That's for the start menu, and probably you know the situation about the taskbar already. I worked tirelessly for 8 months straight reverse engineering their 10 taskbar code and reimplementing it. Once it was done, with 66.1 I decided to pack it together. It got flagged as HackTool, then in 66.3 I excluded it and it appeared to be fine even today. Yeah I know, a lot of work for nothing if we're considering antivirus detections as the end of the world.

Then another problem with how files in ep_setup are packed, ep_setups before 66.x packs every single file as a loose resource, this requires assigning every file a number. There's like 40s of language files for the start menu, so figured that this would be considerably hard. Then I reworked how files are packed, which is packing them in a zip file. It was fine at first with 3 detections, but because of the reworked packing method and the code used to extract them, it caused the binary to be "different" overall. Combined with the behavior that EP drops some files in C:\Windows, this led into Defender's ML models flagging this more likely than usual. 10-15 detections a week after.

The timing for this is, honestly I admit is very bad. At this point Microsoft has been actively flagging new builds for EP for some reason. This makes new x64 builds of ep_setup, no matter who compiled it, keeps getting flagged by their ML models which would raise their humans' attention to flag this as an actual HackTool. The HackTool flags raised by MS eventually spread to other major vendors.

Originally I am not sure why still they started flagging this in February. I feel this is an unfair situation which needs to be disputed. StartAllBack and Windhawk and its mods do similar code patching techniques yet they are still fine.

Funnily enough the DLLs contained within the setup file are not flagged, which is why I am now making a PowerShell-based online installer. It should be way more transparent and less evasive.

Let me know if I missed any detail.

**Comment by Amrsatrio** [comment]
Last but not least my final words on this: Only use this if you trust it and willing to accept the fact that there are many VT detections. I'm not forcing you to use this or to update. If you don't use the 10 start menu or 10 taskbar on any other side than the bottom, you should be good to go with 65.5 (last known good version) and disable EP updates. I did everything that was needed to keep EP's advertised features working as intended on latest builds, and if this is indeed the reality, so be it, it's up to the users.

**Comment by leighss** [comment]
> You can blame Microsoft for doing the entirely unnecessary breakages.

They really are Counts without the 'o'

**Comment by joeysirmons** [comment]
@Amrsatrio thank you very, very much, both for contributing to EP and for your thoughtful answers. I really appreciate it. It's people like you (and the other contributors here) that make open-source software work!

**Comment by Kickskii** [comment]
today it saying the setup is Trojan:Win32/Fauppod!ml

**Comment by KunjanChauhan** [comment]
> And when they flag the PowerShell script, we can easily break the hash by modifying the script

Ideally, there should be a legitimate way to modify the UI, even at the core Explorer level.  Does MSFT provide any API hooks for this though?  The downside is that it would probably require quite a bit of coding.  The upside is that EP is no longer in the the crosshairs.

**Comment by KunjanChauhan** [comment]
I have created a suggestion in MSFT Feedback hub.  This madness by MSFT will only stop if enough people raise a voice.  

If you agree upvote at https://aka.ms/AAs87uy - note this link will open Feedback Hub on Windows.  For some reason, it will not work on an Android device.  Another MSFT architecture blunder!

**Comment by ddbrown30** [comment]
SentinelOne is also flagging the new version as a virus. This is especially a problem for me because this is my work PC and I do not have any control over the virus software i.e. I can't add an exception or remove the quarantine on a specific file. I'll have to revert back to the old version for now.

**Comment by Amrsatrio** [comment]
@ddbrown30 Can you put files in C:\Windows? Or you do not have permissions for it?

**Comment by ddbrown30** [comment]
> @ddbrown30 Can you put files in C:\Windows? Or you do not have permissions for it?

Yeah, I can modify the windows directory. I have full admin access, I just don't have control of the AV.

**Comment by Amrsatrio** [comment]
Probably you can manually put the files extracted from ep_setup from another PC but that's too complicated. Anyways, try my fork and let me know if is flagged as well. Should be good as long as it is not too popular.

https://github.com/Amrsatrio/ExplorerPatcher/releases/tag/22621.3880.66.6_92fce8c

**Comment by ddbrown30** [comment]
Unfortunately, still doesn't work. It's able to download the file fine and I can even scan it with no problems but whatever it's doing during setup causes the AV to see it as a threat and quarantine it.

**Comment by Amrsatrio** [comment]
Yeah manual setup is the only way now for you I think. Can't give you instructions right now but the files are accessible by opening the setup exe in WinRAR or 7-zip.

**Comment by ddbrown30** [comment]
I've got the files unpacked but yeah, not sure where to go from there. Please send me some instructions when you have a moment. Thanks. :)

**Comment by davidfregoli** [comment]
> I have created a suggestion in MSFT Feedback hub. This madness by MSFT will only stop if enough people raise a voice.
> 
> If you agree upvote at https://aka.ms/AAs87uy - note this link will open Feedback Hub on Windows.
> 

does not work with Local account, oh the irony

**Comment by Felessan38** [comment]
> Probably you can manually put the files extracted from ep_setup from another PC but that's too complicated. Anyways, try my fork and let me know if is flagged as well. Should be good as long as it is not too popular.
> 
> https://github.com/Amrsatrio/ExplorerPatcher/releases/tag/22621.3880.66.6_92fce8c

FYI this fork worked for me and was not automatically quarantined.  It did ask if I wanted to keep the file anyway when I downloaded it, which of course I did.  I also sent a report to mark the file safe to Microsoft when it asked if I wanted to keep it.

It seemed to install without issue and no flags from Sentinel One.

With the way this seems to be going, this may be the last update we'll be able to install.

**Comment by Amrsatrio** [comment]
@Felessan38 You submitted it to the form for reporting false positives in their website? It's a bad move actually. By doing so the file will certainly be flagged as HackTool.

**Comment by mxnstrexgl** [comment]
![Screenshot 2024-09-15 144436](https://github.com/user-attachments/assets/2d451c1c-ef67-4f02-86aa-3584165918b8)
Update: Being flagged as a Backdoor by Windows Defender.

**Comment by Kickskii** [comment]
![image](https://github.com/user-attachments/assets/c672731c-202c-4532-8a2f-edaebaf386f8)
today its being detected as a completely new thing wtf is going on

**Comment by ff66theone** [comment]
It first detected Bladabindi when i downloaded it, and when I ran it it, was Contebrew, and when you go to the Microsoft site, for some reasons it says that "Technical details [about the threat behavior] are not available", but it detected it **when i ran it**, not when i downloaded it (good job Microsoft, I have a proof that you're lying :) )

**Comment by pyrates999** [comment]
caused by this: https://en.wikipedia.org/wiki/Heuristic_analysis

**Comment by emapco** [comment]
> Probably you can manually put the files extracted from ep_setup from another PC but that's too complicated. Anyways, try my fork and let me know if is flagged as well. Should be good as long as it is not too popular.
> 
> https://github.com/Amrsatrio/ExplorerPatcher/releases/tag/22621.3880.66.6_92fce8c

VirusTotal flags the fork's [amd64](https://www.virustotal.com/gui/file/f23419768ecc7a906df026ae45d92270544d08f743090186bd7e3f7592e2dd6f?nocache=1) and [arm64](https://www.virustotal.com/gui/file/b41c1e456620908ce0147cb4ffd223688d669bb072e45b557edf9088fbc0846f?nocache=1) executables.

**Comment by Apellamen** [comment]
Win32/Bladabindi!ml

I had to remove EP. What's happening?

**Comment by ff-66** [comment]
The main problem is still the fact that microsoft uses his antivirus to scare "normal users" and make them think it is bad, but legally...

**Comment by KunjanChauhan** [comment]
> > Probably you can manually put the files extracted from ep_setup from another PC but that's too complicated. Anyways, try my fork and let me know if is flagged as well. Should be good as long as it is not too popular.
> > https://github.com/Amrsatrio/ExplorerPatcher/releases/tag/22621.3880.66.6_92fce8c
> 
> VirusTotal flags the fork's [amd64](https://www.virustotal.com/gui/file/f23419768ecc7a906df026ae45d92270544d08f743090186bd7e3f7592e2dd6f?nocache=1) and [arm64](https://www.virustotal.com/gui/file/b41c1e456620908ce0147cb4ffd223688d669bb072e45b557edf9088fbc0846f?nocache=1) executables.

Hi, This worked for me.  Defender flagged a popup stating that source was unknown, but I was able to continue and install.  Thereafter, however, when you go into "EP > Properties > Check for updates", EP still wants to download version main branch *66.6 even though the newly installed fork is at *66.6.  I suspect this is just a dev clean up req'd.  Anyway, downloading the main branch of course results in Defender having a major sissy fit.

It's really nice to have the right click > drop downs back on all the icons and open apps in the task bar...!  Thank you!!!

**Comment by Thebestbmxer** [comment]
Just wanna say the PowerShell commands from the second post by Amrsatrio worked to allow me to run the update from inside EP. Just needed to enable PowerShell as an administrator.

**Comment by pyrates999** [comment]
> > > Probably you can manually put the files extracted from ep_setup from another PC but that's too complicated. Anyways, try my fork and let me know if is flagged as well. Should be good as long as it is not too popular.
> > > https://github.com/Amrsatrio/ExplorerPatcher/releases/tag/22621.3880.66.6_92fce8c
> > 
> > 
> > VirusTotal flags the fork's [amd64](https://www.virustotal.com/gui/file/f23419768ecc7a906df026ae45d92270544d08f743090186bd7e3f7592e2dd6f?nocache=1) and [arm64](https://www.virustotal.com/gui/file/b41c1e456620908ce0147cb4ffd223688d669bb072e45b557edf9088fbc0846f?nocache=1) executables.
> 
> Hi, This worked for me. Defender flagged a popup stating that source was unknown, but I was able to continue and install. Thereafter, however, when you go into "EP > Properties > Check for updates", EP still wants to download version main branch *66.6 even though the newly installed fork is at *66.6. I suspect this is just a dev clean up req'd. Anyway, downloading the main branch of course results in Defender having a major sissy fit.
> 
> It's really nice to have the right click > drop downs back on all the icons and open apps in the task bar...! Thank you!!!
> 
> Below the Virustotal scan also shows much better results:
> 
> ![image](https://private-user-images.githubusercontent.com/8205151/367673812-f90f3d57-27e1-466a-8cda-eac87241d0bf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY0ODUwOTYsIm5iZiI6MTcyNjQ4NDc5NiwicGF0aCI6Ii84MjA1MTUxLzM2NzY3MzgxMi1mOTBmM2Q1Ny0yN2UxLTQ2NmEtOGNkYS1lYWM4NzI0MWQwYmYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MTZUMTEwNjM2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzIxNmZmNzlhYTRmMTU2YjJkNDE3YzkyMWY1NzdiNzBhY2I4MDI2M2YzNTAyMzE0YzE4NjcyMGM4MjBiZjgzZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.-5Uvt-DruPiB6nIUxNKSfPiJys824evx5DWpnyyf9hI)

Please open up new issue for this where it says there is an update to 66.6, but you already have 66.6 installed.

**Comment by pyrates999** [comment]
> Just wanna say the PowerShell commands from the second post by Amrsatrio worked to allow me to run the update from inside EP. Just needed to enable PowerShell as an administrator.
> 
> Edit: Anyone know if there is a .dll to fix the icons?

Open up a new discussion for that.

**Comment by KunjanChauhan** [comment]
> Please open up new issue for this where it says there is an update to 66.6, but you already have 66.6 installed.

done!

**Comment by IanKemp** [comment]
I can understand why antivirus apps picked EP up as suspicious, because it is doing some fairly hinky things to `explorer.exe` and its surrounds, but it really is a shame that not only does this project have to exist to fight back against MS's dumbing-down of its OS - it has to deal with being falsely reported as malicious software.

Thanks for continuing to fight the good fight, EP devs.

**Comment by Hi-ImKyle** [comment]
You can usually get stuff let through Defender if you submit it as a false positive. Is it just the case of Microsoft not caring and/or don't want people using this because it's useful? Or is it just not something EP is bothered to even do every update? If its the latter, I understand.

**Comment by Apellamen** [comment]
> You can usually get stuff let through Defender if you submit it as a false positive. Is it just the case of Microsoft not caring and/or don't want people using this because it's useful? Or is it just not something EP is bothered to even do every update? If its the latter, I understand.

I suppose they just don't want the place where they can push their products can be fixed as the user want. Until someone will fee them for abuse of dominant position they will continue to milk everyone. IT companies like microsoft understand only judge hammers.

Sorry for the offtopic. Good luck to EP.

**Comment by Anamon** [comment]
> You can usually get stuff let through Defender if you submit it as a false positive. Is it just the case of Microsoft not caring and/or don't want people using this because it's useful? Or is it just not something EP is bothered to even do every update? If its the latter, I understand.

I would say the fact that the detection now explicitly includes "ExplorerPatcher" in its name is proof enough that this isn't about Microsoft not caring to fix a false detection, it's them actively and deliberately targetting Explorer Patcher, and trying to get people to stop using it.

The generous interpretation would be that because it performs some deep modification of the shell, they deem it risky, and something that could lead to a lot of load on their customer support (do they have any?) if something breaks. A more cynical interpretation would be that they want to squash its potential of letting users choose how their operating system looks and behaves, including letting them get rid of unsolicited advertisement (in which case I suppose this could be something of interest to anti-trust bodies).

**Comment by 204504bySE** [comment]
Windows Defender should detect Browser:Win64/Chrome  Browser:Win64/Firefox in near future.

**Comment by mayureshs** [comment]
Same here, even  after following the PS script as a first time installer I am not able  to install the program.
![image](https://github.com/user-attachments/assets/3fa8c157-bdd6-4803-8c2e-898abef0082e)

**Comment by ff-66** [comment]
Then disable Smart Screen and real-time protection ?

**Comment by KunjanChauhan** [comment]
> Then disable Smart Screen and real-time protection ? 

That is not recommended as it opens up your system to all sorts of attacks.

**Comment by Thebestbmxer** [comment]
No its not. But you can turn it off, install, then turn back on.
Then use the powershell commands above as well as telling your system it is ok.

**Comment by KunjanChauhan** [comment]
> No its not. But you can turn it off, install, then turn back on. Then use the powershell commands above as well as telling your system it is ok.

Yes that would work in many cases!

**Comment by mayureshs** [comment]
> Then disable Smart Screen and real-time protection ? I did it two days ago on my new PC and this method worked perfectly

Do not have permissions to disable the Smart Screen or real-time protection. Any other options?

**Comment by pyrates999** [comment]
SmartScreen will still prompt you when trying to run an executable that is not signed.  So this is perfectly ok.

**Comment by ErMaqui** [comment]
> I have created a suggestion in MSFT Feedback hub. This madness by MSFT will only stop if enough people raise a voice.
> 
> If you agree upvote at https://aka.ms/AAs87uy - note this link will open Feedback Hub on Windows. For some reason, it will not work on an Android device. Another MSFT architecture blunder!
> 
> Screenshot below:
> 
> ![image](https://private-user-images.githubusercontent.com/8205151/367282278-25622da6-6e87-405a-b8bf-2dc0157e6949.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcwOTMxMTgsIm5iZiI6MTcyNzA5MjgxOCwicGF0aCI6Ii84MjA1MTUxLzM2NzI4MjI3OC0yNTYyMmRhNi02ZTg3LTQwNWEtYjhiZi0yZGMwMTU3ZTY5NDkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjNUMTIwMDE4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzdlY2YyMDRjZWQ1NGE4MjJmYWVjYzVhY2RhMjcyZDcxYjUwODViY2Q2Yzk2ODUwZTFjYzU0MDI1OTk2N2RmYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.L4MoscvnpKHk88XCTGD6A_K6HzUyo0MydDvADzf2xqM)

![image](https://github.com/user-attachments/assets/7f4602be-b3b2-4878-924c-2df0d53c8089)

I think M$ closed your request...

**Comment by KunjanChauhan** [comment]
> > I have created a suggestion in MSFT Feedback hub. This madness by MSFT will only stop if enough people raise a voice.
> > If you agree upvote at https://aka.ms/AAs87uy - note this link will open Feedback Hub on Windows. For some reason, it will not work on an Android device. Another MSFT architecture blunder!
> > Screenshot below:
> > ![image](https://private-user-images.githubusercontent.com/8205151/367282278-25622da6-6e87-405a-b8bf-2dc0157e6949.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcwOTMxMTgsIm5iZiI6MTcyNzA5MjgxOCwicGF0aCI6Ii84MjA1MTUxLzM2NzI4MjI3OC0yNTYyMmRhNi02ZTg3LTQwNWEtYjhiZi0yZGMwMTU3ZTY5NDkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjNUMTIwMDE4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzdlY2YyMDRjZWQ1NGE4MjJmYWVjYzVhY2RhMjcyZDcxYjUwODViY2Q2Yzk2ODUwZTFjYzU0MDI1OTk2N2RmYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.L4MoscvnpKHk88XCTGD6A_K6HzUyo0MydDvADzf2xqM)
> 
> ![image](https://private-user-images.githubusercontent.com/8928758/369863715-7f4602be-b3b2-4878-924c-2df0d53c8089.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcwOTU0NjksIm5iZiI6MTcyNzA5NTE2OSwicGF0aCI6Ii84OTI4NzU4LzM2OTg2MzcxNS03ZjQ2MDJiZS1iM2IyLTQ4NzgtOTI0Yy0yZGYwZDUzYzgwODkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjNUMTIzOTI5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzUwZDFjOGI1MzBjNjMwYjIyYWQwYmViNTVmMTg4ZWFjMTAwYjA3OTVjNmQ0YzJjOGM2M2JmNzViMDQ3ZDEzOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.8a1g2xe_ucVIl5F8p654m0G845fjIEOX6gUSTHcP1J4)
> 
> I think M$ closed your request...

I just checked and its still open.  There are 22 upvotes.  If there are more experiences like yours then clearly something may be amiss .....! 

In any case, this requested change will not be forthcoming from MSFT through just one post on the feedback hub.  I think we need a lot more upvotes and overall noise through various other feedback pipes.  It's not impossible to imagine MSFT not changing their point of view!

**Comment by osde8info** [comment]
windows 11 edge download says both exporerpatcher 65.5 and LATEST are viruses

tried to upvote via your link but got 

![winfeed](https://github.com/user-attachments/assets/371707bf-9601-4f0f-b934-ed5667c2b8db)

had to upvote via this link instead https://aka.ms/AAse02b

**Comment by osde8info** [comment]
also upvoted native windows 11 cascade issue https://aka.ms/AAo78gw

**Comment by KunjanChauhan** [comment]
> windows 11 edge download says both exporerpatcher 65.5 and LATEST are viruses
> 
> tried to upvote via your link but got
> 
> ![winfeed](https://private-user-images.githubusercontent.com/51541/369988431-371707bf-9601-4f0f-b934-ed5667c2b8db.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcxMTcyMjYsIm5iZiI6MTcyNzExNjkyNiwicGF0aCI6Ii81MTU0MS8zNjk5ODg0MzEtMzcxNzA3YmYtOTYwMS00ZjBmLWI5MzQtZWQ1NjY3YzJiOGRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIzVDE4NDIwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYzYjdlOTQxYmQxMTczZGUzMTBjNzJlODZhYmU4ODY1ZTM5NmU3NDk0MWJjZDZjNmQ4MmIwYWY0MTliZDA3OGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Bf4btC_gf_UmUeV6jCl9PnfIkBDgHhYzxv_kgHxjijI)
> 
> had to upvote via this link instead https://aka.ms/AAse02b

It seems that the only way to access the suggestion is from within Windows/Feedback Hub app.  I am unable to access via a browser or non-windows device such as my android phone.  What can I say....!!!!!

**Comment by rodolfomachado** [comment]
Here it says that my account don't have rights to vote

**Comment by ErMaqui** [comment]
> > windows 11 edge download says both exporerpatcher 65.5 and LATEST are viruses
> > tried to upvote via your link but got
> > ![winfeed](https://private-user-images.githubusercontent.com/51541/369988431-371707bf-9601-4f0f-b934-ed5667c2b8db.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcxMTcyMjYsIm5iZiI6MTcyNzExNjkyNiwicGF0aCI6Ii81MTU0MS8zNjk5ODg0MzEtMzcxNzA3YmYtOTYwMS00ZjBmLWI5MzQtZWQ1NjY3YzJiOGRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIzVDE4NDIwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYzYjdlOTQxYmQxMTczZGUzMTBjNzJlODZhYmU4ODY1ZTM5NmU3NDk0MWJjZDZjNmQ4MmIwYWY0MTliZDA3OGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Bf4btC_gf_UmUeV6jCl9PnfIkBDgHhYzxv_kgHxjijI)
> > had to upvote via this link instead https://aka.ms/AAse02b
> 
> It seems that the only way to access the suggestion is from within Windows/Feedback Hub app. I am unable to access via a browser or non-windows device such as my android phone. What can I say....!!!!!

Link opens automatically the feedback hub app... but the result are the same.

**Comment by pyrates999** [comment]
> > windows 11 edge download says both exporerpatcher 65.5 and LATEST are viruses
> > tried to upvote via your link but got
> > ![winfeed](https://private-user-images.githubusercontent.com/51541/369988431-371707bf-9601-4f0f-b934-ed5667c2b8db.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcxMTcyMjYsIm5iZiI6MTcyNzExNjkyNiwicGF0aCI6Ii81MTU0MS8zNjk5ODg0MzEtMzcxNzA3YmYtOTYwMS00ZjBmLWI5MzQtZWQ1NjY3YzJiOGRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIzVDE4NDIwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYzYjdlOTQxYmQxMTczZGUzMTBjNzJlODZhYmU4ODY1ZTM5NmU3NDk0MWJjZDZjNmQ4MmIwYWY0MTliZDA3OGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Bf4btC_gf_UmUeV6jCl9PnfIkBDgHhYzxv_kgHxjijI)
> > had to upvote via this link instead https://aka.ms/AAse02b
> 
> It seems that the only way to access the suggestion is from within Windows/Feedback Hub app. I am unable to access via a browser or non-windows device such as my android phone. What can I say....!!!!!

see this comment: https://github.com/valinet/ExplorerPatcher/issues/3228#issuecomment-2273435834

**Comment by lucianoGG** [comment]
Windows 11 24h2 without issues, Kaspersky didn't detect anything

Version 22621.3880.66.6

**Comment by pyrates999** [comment]
The powershell script must run with elevated privileges.

**Comment by Shane32** [comment]
Have any devs attempted to communicate with Microsoft via https://www.microsoft.com/en-us/wdsi/filesubmission or reviewed the criteria for classification as PUA to see if perhaps some additional notices during installation are all that is necessary (see https://learn.microsoft.com/en-us/defender-xdr/criteria ).  I sorta doubt that the feedback hub will do anything.  Just wondering...

**Comment by Amrsatrio** [comment]
Yes Valinet did.

https://github.com/valinet/ExplorerPatcher/issues/3670#issuecomment-2358741849

**Comment by MonoxideXZ** [comment]
GUYS BE CAREFUL EXPLORER PATCHER IS A GAME HACK!!!!!!!!11!!!! (joke)

**Comment by BenReilly** [comment]
Despite the above powershell commands executing successfully, Windows is still blocking me from updating Windows 11, blaming ExplorerPatcher. Do you think it would be successful if I uninstalled EP, updated, and then resinstalled EP?

**Comment by pyrates999** [comment]
> Despite the above powershell commands executing successfully, Windows is still blocking me from updating Windows 11, blaming ExplorerPatcher. Do you think it would be successful if I uninstalled EP, updated, and then resinstalled EP?

You always are supposed to uninstall EP before upgrading between windows 11 releases.

**Comment by ff66theone** [comment]
Yeah unfortunately Microsoft blocks updates for EP users...

**Comment by KunjanChauhan** [comment]
> > Despite the above powershell commands executing successfully, Windows is still blocking me from updating Windows 11, blaming ExplorerPatcher. Do you think it would be successful if I uninstalled EP, updated, and then resinstalled EP?
> 
> You always are supposed to uninstall EP before upgrading between windows 11 releases.

I have NEVER had to uninstall and reinstall...!

**Comment by KunjanChauhan** [comment]
I just installed the latest MSFT updates followed by an auto-prompt EP update (v...67.1).  Everything was fine.  

The image shows which MSFT updates were installed.

![image](https://github.com/user-attachments/assets/4a6b050a-b2e2-476b-9a39-6a443c06e8a4)

**Comment by pyrates999** [comment]
We're referring to when you do an upgrade from 23h2 to 24h2, that is when you should uninstall EP first.  Not for regular windows updates.

**Comment by ErMaqui** [comment]
Simple question.

If I remember correctly, one of the "potential problems" is the inclusion on the installer of some .dll files deleted by the OS.
If you can download it automatically from the app (as the app do with debug symbols). maybe the hack notice will dissapear?

Also, if the download per se is a problem because defender will block it, you can prevent the user during installation (if defender reverts the decision over the installer) to do the requiered configuration on windows defender, or even, do the configuration with the installer if the user approves.

**Comment by Amrsatrio** [comment]
> If I remember correctly, one of the "potential problems" is the inclusion on the installer of some .dll files deleted by the OS.
If you can download it automatically from the app (as the app do with debug symbols). maybe the hack notice will dissapear?

Total rewrite of the setup is needed if we need to add some UI there. I don't have time and motivation for it yet. For the downloads, I've remembered that there were reports where when EP wasn't installed without an Internet connection, the 10 start menu was broken to a certain extent due to some missing restoration files. Though in the future I'm considering extending the symbol download feature to download binaries as well during runtime. There will be no extended black screens (the StartUI.dll is 10MB), although the user has to wait couple minutes till it finishes downloading especially on bad internet conditions.

@ff66theone Nice one, I'm not letting EP becoming real malware 🤪

**Comment by Amrsatrio** [comment]
Yes, of course, I feel this is required for downloads that may take a long time.

**Comment by ff66theone** [comment]
> @ff66theone Nice idea actually, but I'm not letting EP becoming real malware 🤪

I know that this technique is questionable but it is working well, also i don't want XORed code, alone it works already, and the only thing to do is asking for consent or implementing a toggle for that functionality. I don't want that feature to be hidden, i want it to be shown so users don't think it is here  for bad reasons.
